### PR TITLE
Made the test for the navbar header element less brittle. See other changes below.

### LINKED ISSUE #74: Flexbox solution fails unit tests - tech docs project
The testing for this doesn't allow for flexbox configuration, since the contents of the flexbox need to be wrapped in a div with position set to fixed for it to work, and this causes the following conditions to fail:

CONTENT
 9. The first element within the navbar should be a <header> which contains text that describes the topic of the technical documentation.
(failed because first element was the div mentioned above)

LAYOUT 
1. On regular sized devices (laptops, desktops), the element with id="navbar" should be shown on the left side of the screen and should always be visible to the user.
failed with "AssertionError: expected -1000 to be close to 0 +/- 10"

If your intention is to get people to use another approach, like fixed and absolute positioning for the two main sections - kindly ignore this issue!

#### Browser Information
Using iMac w/Chrome browser Version 56.0.2924.87 (64-bit)

#### Your Code / Link to Your Pen
http://codepen.io/sue888/pen/OpRoqB

### PR DESCRIPTION
**Pull request by tchaffee**
- Added a test to make sure the navbar header comes before the links.
- Added chai assert messages to make is clearer where the layout tests fail.
- Note there are some whitespace changes to remove trailing whitespaces.

#### Pre-Submission Checklist

- [x] Your pull request targets the `master` branch of freeCodeCamp/testable-projects-fcc
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](http://forum.freecodecamp.com/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [x] Your changes have been tested either locally or using a newly created CDN based on your fork's testable-projects-fcc/build/bundle.js file
- [x] Working changes have been added to the build by running `npm run build`

#### Type of Change

- [x] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)

#### Checklist:

- [x] Tested changes locally.
- [x] Closes currently open issue (replace XXX with an issue no): Closes #74 

#### Description
The navbar section can now include containers outside the header element and the tests will still pass, making the tests less brittle. Also added some Chai messages to make it clearer where tests fail.

### CODE REVIEW

File: src/project-tests/technical-docs-tests.js [hunk 1]
```diff
@@ -93,11 +93,11 @@ export default function createTechnicalDocsPageTests() {
...
             });
 
-            it('9. The first element within the navbar should be a <header> which contains text that describes the topic ' +
+            it('9. The navbar element should contain one <header> element which contains text that describes the topic ' +
                 'of the technical documentation.',
                 function() {
-                    FCC_Global.assert.strictEqual(document.getElementById('navbar').firstElementChild.nodeName, 'HEADER', 'The first ' +
-                        'element within #navbar should be a <header> ');
+                    FCC_Global.assert.strictEqual(document.getElementById('navbar').getElementsByTagName('header').length, 1, 'The navbar ' +
+                        'element should contain ONE <header> element.');
```
Comment (no-stack-dub-sack) [inline src/project-tests/technical-docs-tests.js]: @tchaffee if you're updating this PR anyway, I would not capitalize the word "one". I think it looks a little funky. Normal casing should be fine.

File: src/project-tests/technical-docs-tests.js [hunk 2]
```diff
@@ -149,11 +165,12 @@ export default function createTechnicalDocsPageTests() {
...
-                FCC_Global.assert.approximately(navbar.getBoundingClientRect().left, 0, 10);
-                FCC_Global.assert.approximately(navbar.getBoundingClientRect().top, 0, 10);
+                FCC_Global.assert.approximately(navbar.getBoundingClientRect().left, 0, 10, 'Left of bounding rectangle is not correct.');
+                FCC_Global.assert.approximately(navbar.getBoundingClientRect().top, 0, 10, 'Top of bounding rectangle is not correct.');
                 window.scroll(0, 1000);
-                FCC_Global.assert.approximately(navbar.getBoundingClientRect().left, 0, 10);
-                FCC_Global.assert.approximately(navbar.getBoundingClientRect().top, 0, 10);
+                boundingRect = navbar.getBoundingClientRect();
+                FCC_Global.assert.approximately(navbar.getBoundingClientRect().left, 0, 10, 'After scroll: Left of bounding rectangle is not correct.');
+                FCC_Global.assert.approximately(navbar.getBoundingClientRect().top, 0, 10, 'After scroll: Top of bounding rectangle is not correct.');
```
Comment (no-stack-dub-sack) [inline src/project-tests/technical-docs-tests.js]: These are nice additions.

File: src/project-tests/technical-docs-tests.js [hunk 3]
```diff
@@ -114,7 +114,23 @@ export default function createTechnicalDocsPageTests() {
...
 
-            it('11. Each element with the class of "nav-link" should contain text that corresponds to the <header> text ' +
+            it('11. The <header> element in the navbar must come before any link (<a>) elements in the navbar.',
+                function() {
+                  const navLinks = document.querySelectorAll('#navbar a.nav-link');
+                  const header = document.querySelectorAll('#navbar header').item(0);
+                  let allAbove = true;
+
+                  navLinks.forEach(navLink => {
+                    if ( ! (header.compareDocumentPosition(navLink) & Node.DOCUMENT_POSITION_FOLLOWING)) {
```
Comment (no-stack-dub-sack) [inline src/project-tests/technical-docs-tests.js]: I would keep the `!`, right up against the value you are trying to negate, e.g. `!true` - more commonly written this way. 

Nice use of the bitwise `&` operator! I've very rarely seen a good use case for it. Looks like you read the docs carefully for this. Nice job.

Comment (tchaffee) [inline src/project-tests/technical-docs-tests.js]: @no-stack-dub-sack Oops, old habit: the space between the paren and bang is something I insist on at my own company. The `!` operator is harder to miss when it isn't shoved against the paren, but I do understand that style is not very common (yet  ;-)). I will get this changed and resolve the merge conflict.

Thanks for your timely review!

File: src/project-tests/technical-docs-tests.js (hunk 1 above)
Comment (tchaffee) [inline src/project-tests/technical-docs-tests.js]: Ok, fixed.

**Comment by tchaffee** [comment]
@no-stack-dub-sack Could you just review my changed files again carefully? I think I rebased correctly, but there are some changes in my files that I was not expecting. Like the lines with double quote changed to single quote. I'm not sure why that is happening. Maybe my editor is automatically changing it or ESLint changed it since I was experimenting with ESLint. Annoying that it clutters up this commit, but I think the changes do make the code more consistent, so hopefully they are all ok? Thanks.

**Comment by no-stack-dub-sack** [comment]
@tchaffee Looks good to me! I'm not worried about the quotes, that's all good. Great job with this one!

**Pull request merged**